apis/workloads/v1alpha1/replicatedstatemachine_types.go

```diff
@@ -54,28 +74,44 @@ type ReplicatedStateMachineSpec struct {
 	// +optional
 	VolumeClaimTemplates []corev1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty"`
 
+	// podManagementPolicy controls how pods are created during initial scale up,
+	// when replacing pods on nodes, or when scaling down. The default policy is
+	// `OrderedReady`, where pods are created in increasing order (pod-0, then
+	// pod-1, etc) and the controller will wait until each pod is ready before
+	// continuing. When scaling down, the pods are removed in the opposite order.
+	// The alternative policy is `Parallel` which will create pods in parallel
+	// to match the desired scale without waiting, and on scale down will delete
+	// all pods at once.
+	// +optional
+	PodManagementPolicy appsv1.PodManagementPolicyType `json:"podManagementPolicy,omitempty"`
+
+	// updateStrategy indicates the StatefulSetUpdateStrategy that will be
+	// employed to update Pods in the RSM when a revision is made to
+	// Template.
+	// UpdateStrategy.Type will be set to appsv1.OnDeleteStatefulSetStrategyType if MemberUpdateStrategy is not nil
+	UpdateStrategy appsv1.StatefulSetUpdateStrategy `json:"updateStrategy,omitempty"`
+
 	// Roles, a list of roles defined in the system.
-	// +kubebuilder:validation:Required
-	Roles []ReplicaRole `json:"roles"`
+	// +optional
+	Roles []ReplicaRole `json:"roles,omitempty"`
 
 	// RoleObservation provides method to observe role.
-	// +kubebuilder:validation:Required
-	RoleObservation RoleObservation `json:"roleObservation"`
+	// +optional
+	RoleObservation *RoleObservation `json:"roleObservation,omitempty"`
```

Comment:
Probe?

Reply:
done in: [f7f09bf](https://github.com/apecloud/kubeblocks/pull/4086/commits/f7f09bff2f1a44af89ab5762a97d6abecf465ccc)